src/target/riscv/riscv-013.c

```diff
@@ -565,6 +600,10 @@ static int dmi_op_timeout(struct target *target, uint32_t *data_in,
 		bool *dmi_busy_encountered, int dmi_op, uint32_t address,
 		uint32_t data_out, int timeout_sec, bool exec, bool ensure_success)
 {
+	uint32_t address_with_dbgbase = address;
+	if (target->dbgbase_set)
+		address_with_dbgbase += target->dbgbase;
```

Comment:
This construct where you check for dbg_base_set and then maybe add something is kind of clunky.

When it's not set, can we simply ensure target->dbgbase is 0, and then then all the add/subtract will simply be a nop and leave the code much cleaner.

Reply:
https://github.com/riscv/riscv-openocd/blob/4be6a5683f9de5705c5d8a28f4ed9ada2250dccb/src/target/target.c#L6229

 target->dbgbase should be 0 as target is init by calloc